aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -214,47 +192,30 @@ Tensor& normal_impl_(Tensor& self, double mean, double std, c10::optional<Genera
 
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, const Tensor& mean, double std, c10::optional<Generator> gen) {
+  CHECK_NORMAL_SIZES(output, mean);
+  CHECK_NORMAL_STD(std);
   normal_impl_<normal_kernel, RNG>(output, 0, std, gen);
   output.add_(mean);
   return output;
 }
 
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, double mean, const Tensor& std, c10::optional<Generator> gen) {
-  TORCH_CHECK(!std.is_complex(), "normal expects standard deviation to be non-complex");
-  TORCH_CHECK(
-    std.min().ge(0).item<bool>(),
-    "normal expects all elements of std >= 0.0");
+  CHECK_NORMAL_SIZES(output, std);
+  CHECK_NORMAL_TENSOR_STD(std);
   normal_impl_<normal_kernel, RNG>(output, 0, 1, gen);
   auto mean_tensor = at::full({}, mean, output.options());
-  // CUDA NB: addcmul_out copies the tensor to be added into the output.
-  // Please look at aten/src/THC/generic/THCTensorMathPointwise.cu
-  // The previous function here was addcmul_out(output, mean_tensor, output, std, 1);
-  // The third argument is not a constant reference and hence the samples in output are overwritten.
-  // Consequently, the computation performed is mean_tensor + mean_tensor * std instead of mean_tensor + output * std
   output.mul_(std).add_(mean_tensor);
   return output;
 }
 
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
-  TORCH_CHECK(!std.is_complex(), "normal expects standard deviation to be non-complex");
-  TORCH_CHECK(
-    std.numel() == 0 || std.min().ge(0).item<bool>(),
-    "normal expects all elements of std >= 0.0");
-  bool is_deprecated_th_impl = resize_output_for_normal(output, mean, std);
+  CHECK_NORMAL_SIZES(output, mean);
+  CHECK_NORMAL_SIZES(output, std);
+  CHECK_NORMAL_TENSOR_STD(std);
```

Comment:
(pretty much) no checks should happen in the `_impl` side of things. They already happen in the `meta` part, so this should probably go.

Reply:
see above